README.md

```diff
@@ -1,2 +1,257 @@
+Humble still supports Gazebo Classic, but Jazzy (Ubuntu 24.04) does not.
+Migration will be needed when moving to Jazzy.
+
+## Building Humble on Ubuntu 20.04
+Install a few dependencies:
+
+```bash
+sudo apt install -y ros-dev-tools python3-rosinstall-generator
+```
+
+Run these commands to prepare the workspace with everything you need:
+
+```bash
+mkdir -p ros2_humble_ws/src
+cd ros2_humble_ws
+
+rosinstall_generator --deps --rosdistro humble desktop_full \
+    launch_xml \
+    launch_yaml \
+    launch_testing \
+    launch_testing_ament_cmake \
+    demo_nodes_cpp \
+    demo_nodes_py \
+    example_interfaces \
+    camera_calibration_parsers \
+    camera_info_manager \
+    cv_bridge \
+    v4l2_camera \
+    vision_opencv \
+    vision_msgs \
+    image_geometry \
+    image_pipeline \
+    image_transport \
+    compressed_image_transport \
+    compressed_depth_image_transport \
+    rosbag2_storage_mcap \
+    rtabmap \
+    rtabmap_ros \
+    diagnostics \
+    turtlebot3_gazebo \
+    turtlebot3_description \
+    turtlebot3_navigation2 \
+    gazebo_ros_pkgs \
+    joint_state_publisher_gui \
+    rqt_tf_tree \
+> ros2.humble.opentera_webrtc_ros.rosinstall
+
+sed -i '$d' ros2.humble.opentera_webrtc_ros.rosinstall
+
+cat <<EOF >> ros2.humble.opentera_webrtc_ros.rosinstall
+- git:
+    local-name: cv_camera
+    uri: https://github.com/Kapernikov/cv_camera.git
+    version: master
+- git:
+    local-name: xtl
+    uri: https://github.com/xtensor-stack/xtl.git
+    version: 0.7.2
+- git:
+    local-name: xtensor
+    uri: https://github.com/xtensor-stack/xtensor.git
+    version: 0.23.10
+- git:
+    local-name: xsimd
+    uri: https://github.com/xtensor-stack/xsimd.git
+    version: 7.6.0
+EOF
+```
+This combines a `rosinstall_generator` call to get all repos from the `desktop_full` pack,and additionnal dependencies. It also adds the `cv_camera` package, which was ported to ROS2 in a fork that is not available via `rosinstall_generator`, and the `xtl`, `xtensor` and `xsimd` C++ libraries using the versions they have on Ubuntu 22.04. for maximum compatibility.
+
+Then run these commands to clone all the repos and install their dependencies:
+```bash
+vcs import src < ros2.humble.opentera_webrtc_ros.rosinstall
+rosdep install --from-paths src --ignore-src -y --skip-keys "fastcdr rti-connext-dds-6.0.1 urdfdom_headers xsimd xtensor test_pluginlib" --rosdistro humble
+```
+
+You will need to apply a few patches as shown [here](https://github.com/introlab/t-top/blob/ros2-migration/tools/setup_scripts/ros2_humble_install.sh#L241-L242). The patch files are [here (raw libg2o)](https://raw.githubusercontent.com/introlab/t-top/ros2-migration/tools/setup_scripts/patch/libg2o.patch) and [here (raw octomap_msgs)](https://raw.githubusercontent.com/introlab/t-top/ros2-migration/tools/setup_scripts/patch/octomap_msgs.patch).
+
+Create a file named `colcon_defaults.yaml` in the root of the workspace with the following content:
```

Comment:
Can we add optional optimisation flags for ARM  ? Can we add optional optimisation flags to build the  realsense driver with CUDA support ?

Reply:
I could just link to the install script for T-Top and explain that there are optimization flags there for the people interested?